docs/RFCS/sql_batching.md

```diff
@@ -0,0 +1,669 @@
+in the statement independence rules will produce the empty set.
+
+Perhaps more interestingly, we can also reason that two statements that perform mutations
+(INSERT, UPDATE, DELETE, etc.) but that touch disjoint sets of data are also independent
+of one another. It is important to note here that the set of data that a statement _touches_
+includes any that the statement mutates and any that the statement reads while performing
+validation and filtering.
+
+Taking this a step further, we can use our statement independence rules to find that two
+mutating statements can be independent even if their touched data sets overlap, as long
+as only their read-sets overlap. A practical example of this result is a pair of statements
+that insert rows into two different tables, but each validates a foreign key in the same
+third table. These two statements are independent because only their read-sets overlap.
+
+Finally, we can reason that if there is overlap between the touched sets of data between
+any two statements where at least one of the two performs mutation in the overlapping region,
+then the statements are not independent.
+
+#### Read and Write-Set Entries
+
+Up until this point, we have been intentionally vague about the "data" stored in the read
+and write-sets of statements. The reason for this is that the definition of "independence"
+assumes a more general notion of data than just that stored in the rows of a SQL table.
+Instead, this data can include any physical or log­i­cal information that makes up the
+external-facing state of the database. Using this definition, all of the following can fall
+into the read or write-sets of statements:
+
+- Cells/rows in a table
+- Table indexes
+- Table constraint state
+- Table schemas
+- Table and database privileges
+- Inter-statement constraint state
+
+Defining these as different data values that can all be part of statements read and
+write-sets creates some interesting results. For instance, it creates a dependence between
+any statement that implicitly behaves based on some meta-level data and any statement that
+explicitly modifies that meta-level data. An example of this is a query statement to a table
+and a subsequent schema change statement on that table. In this example, the initial query's
+execution semantics are dictated by the table schema, so we add this data to the query's
+read set. Later, the other statement modifies this schema, so we add this data to the query's
+write set. Because the read-set of the first query intersects with the write-set of the second
+query, the two statements must be dependent. It becomes clear that even simple queries have a
+number of implicit data accesses, which need to be accounted for accurately in order to safely
+decide upon statements independence.
+
+#### Examples
+
+```
+SELECT * FROM a
+SELECT * FROM b
+SELECT * FROM a
+```
+All three statements are **independent** because they have empty write-sets.
```

Comment:
This one seems right to me. All three statements are read-only, so they have intersecting read-sets but empty write-sets.